frontend/components/puzzle3.jsx

```diff
@@ -1,27 +1,73 @@
 import React from 'react';
-
 import PuzzleHeader from './puzzleHeader';
+import { AnswerAwareDiv } from './puzzle.jsx';
 
 // Answer: Your name
-
+// Answer: [5,4,3,2,1]
 class Puzzle3 extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = {
+      combo : ['','','','','']
+    };
+  }
+
+
+  handleClick(input) {
+    var temparray = this.state.combo;
+      var flag = 'full';
+      for (var i = 0; i <= temparray.length; i++) {
+        if ( temparray[i] === '') {
+          temparray.splice(i,1,input);
+          this.setState({ combo: temparray });
+          flag = 'good';
+          break;
+        }
+      }
+      if (flag === 'full') {
+        console.log('too many numbers');
+      }
+    }
+  submitanswer() {
+    var userinput = this.state.combo;
+    this.props.send('submit', { puzzle: '3', answer: userinput });
+
+  }
+
+  handleClear() {
+    this.setState({ combo:['','','','',''] });
+    this.props.reset();
+  }
+
   render() {
+    const pw = this.state.combo;
     return (
       <div>
         <PuzzleHeader title="Puzzle Three" />
-        <form id="puzzle3-form">
-          <p>
-            What belongs to you but other people use it more than you?
-          </p>
-          <p>
-            <input type="text" name="puzzleAnswer" autoComplete="off" placeholder="Type here" />
-          </p>
-
-          <div>
-            <button type="submit">submit</button>
-          </div>
-        </form>
-
+        <div className="riddle">
+          <p>Some riddle pertaining to the pictures below</p>
+          <p>make the correct selections are you&apos;ll best this foe</p>
+        </div>
+        <div>
+          {pw.map(pw => <AnswerAwareDiv correct={this.props.correct}> {pw} </AnswerAwareDiv>)})
+        </div>
+        <div className="image-keypad-container">
+          <button id="keypad-1" className="keypad-image" onClick={()=> this.handleClick(1)}>1</button>
```

Comment:
doing this recreates the function after every render, so lets have handleClick return the function:
```
handleClick(input) {
  return () => {
    <use input here>
  }
}
```